--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -1426,6 +1426,7 @@
           'sources': [
             '<(V8_ROOT)/src/base/platform/platform-vibeos-memory.cc',
             '<(V8_ROOT)/src/base/platform/platform-vibeos-time.cc',
+            '<(V8_ROOT)/src/base/platform/platform-vibeos-stack.cc',
           ],
         }],
         ['is_posix or is_fuchsia or (_toolset=="host" and host_os in "mac linux")', {
