--- a/src/node_internals.h
+++ b/src/node_internals.h
@@ -327,7 +327,9 @@
 // Functions defined in node.cc that are exposed via the bootstrapper object
 
 #if defined(__POSIX__) && !defined(__ANDROID__) && !defined(__CloudABI__)
+#if !defined(__vibeos__)
 #define NODE_IMPLEMENTS_POSIX_CREDENTIALS 1
+#endif
 #endif  // defined(__POSIX__) && !defined(__ANDROID__) && !defined(__CloudABI__)
 
 namespace credentials {
--- a/src/node_credentials.cc
+++ b/src/node_credentials.cc
@@ -72,7 +72,7 @@
 // process does not have any capabilities set and the process is running as
 // setuid root then lookup will not be allowed.
 bool SafeGetenv(const char* key, std::string* text, Environment* env) {
-#if !defined(__CloudABI__) && !defined(_WIN32)
+#if !defined(__CloudABI__) && !defined(_WIN32) && !defined(__vibeos__)
 #if defined(__linux__)
   if ((!HasOnly(CAP_NET_BIND_SERVICE) && linux_at_secure()) ||
       getuid() != geteuid() || getgid() != getegid())
